src/lib/y2partitioner/sequences/add_partition.rb

```diff
@@ -0,0 +1,121 @@
+require "yast"
+require "ui/sequence"
+require "y2partitioner/device_graphs"
+require "y2partitioner/dialogs/partition_size"
+require "y2partitioner/dialogs/partition_type"
+
+Yast.import "Wizard"
+
+module Y2Partitioner
+  module Sequences
+    # formerly EpCreatePartition, DlgCreatePartition
+    class AddPartition < UI::Sequence
+      include Yast::Logger
+      # @param disk_name [String]
+      def initialize(disk_name)
+        textdomain "storage"
+        @disk_name = disk_name
+        # collecting params of partition to be created?
+        @ptemplate = Struct.new(:type, :region).new
+      end
+
+      def disk
+        dg = DeviceGraphs.instance.current
+        Y2Storage::Disk.find_by_name(dg, @disk_name)
+      end
+
+      def run
+        sequence_hash = {
+          "ws_start"      => "preconditions",
+          "preconditions" => { next: "type" },
+          "type"          => { next: "size" },
+          "size"          => { next: "role", finish: :finish },
+          "role"          => { next: "format_mount" },
+          "format_mount"  => { next: "password", finish: :finish },
+          "password"      => { finish: :finish }
+        }
+
+        sym = nil
+        DeviceGraphs.instance.transaction do
+          sym = wizard_next_back do
+            super(sequence: sequence_hash)
+          end
+
+          sym == :finish
+        end
+        sym
+      end
+
+      # FIXME: move to Wizard
+      def wizard_next_back(&block)
+        Yast::Wizard.OpenNextBackDialog
+        block.call
+      ensure
+        Yast::Wizard.CloseDialog
+      end
+
+      def preconditions
+        pt = partition_table(disk)
+        slots = pt.unused_partition_slots
+        if slots.empty?
+          Yast::Popup.Error(
+            Yast::Builtins.sformat(
+              _("It is not possible to create a partition on %1."),
+              @disk_name
+            )
+          )
+          return :back
+        end
+        @slots = slots
+        :next
+      end
+      skip_stack :preconditions
+
+      def type
+        Dialogs::PartitionType.run(disk, @ptemplate, @slots)
+      end
+
+      def size
+        Dialogs::PartitionSize.run(disk, @ptemplate, @slots)
+      end
+
+      def role
+        log.info "TODO: Partition ROLE dialog"
+        :next
+      end
+      skip_stack :role
+
+      def format_mount
+        ptable = disk.partition_table
+        name = next_free_primary_partition_name(@disk_name, ptable)
```

Comment:
Ah, yes: this is missing a big FIXME. Which reminds me of what you already have as a FIXME in yast-storage-ng: when creating a partition, the name should be a return value, not an input argument.

As for being lost in a forest of widgets, yes, that is a real problem, and I was already thinking of making a "map" for new contributors.